gpm_api/tests/dataset/test_granule.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+import xarray as xr
+from datatree import DataTree
+
+from gpm_api.dataset import granule
+
+
+# Tests for public functions ###################################################
+
+
+def test_get_variables():
+    """Test get_variables"""
+
+    da = xr.DataArray()
+    dataset = xr.Dataset(data_vars={"var_1": da, "var_2": da})
+    expected_variables = ["var_1", "var_2"]
+    returned_variables = granule.get_variables(dataset)
+    assert returned_variables == expected_variables
+
+
+def test_get_variables_dims():
+    """Test get_variables_dims"""
+
+    array_1 = np.zeros(shape=(3, 3))
+    array_2 = np.zeros(shape=(3, 3))
+    dataarray_1 = xr.DataArray(array_1, dims=["dim_1", "dim_2"])
+    dataarray_2 = xr.DataArray(array_2, dims=["dim_2", "dim_3"])
+    dataset = xr.Dataset(data_vars={"var_1": dataarray_1, "var_2": dataarray_2})
+
+    expected_variables_dims = ["dim_1", "dim_2", "dim_3"]
+    returned_variables_dims = granule.get_variables_dims(dataset)
+    assert np.array_equal(returned_variables_dims, expected_variables_dims)
+
+    # Check dataset with no variables
+    dataset = xr.Dataset()
+    expected_variables_dims = []
+    returned_variables_dims = granule.get_variables_dims(dataset)
+    assert returned_variables_dims == expected_variables_dims
+
+
+def test_unused_var_dims_and_remove():
+    """Test unused_var_dims and remove_unused_var_dims"""
+
+    array = np.zeros(shape=(3,))
+    dataarray = xr.DataArray(array, dims=["used_dim"])
+    dataset = xr.Dataset(data_vars={"var": dataarray})
+    dataset = dataset.expand_dims(dim=["unused_dim"])
+    # TODO: this does not work, it adds the dimension to all variables. How can this be done?
```

Comment:
I see. I guess to create the test case, you need to create two data arrays, then to one of them you call the `expand_dims`, then you `xr.merge`, and then you drop the variable with the `"unused dim"`.
It might be that you need to define the dimension coordinate to avoid that by dropping the variable it drops also the dimension (because completely unused)